stan/math/prim/mat.hpp

```diff
@@ -46,6 +46,8 @@
 #include <stan/math/prim/mat/err/constraint_tolerance.hpp>
 #include <stan/math/prim/mat/err/validate_non_negative_index.hpp>
 
+#include <stan/math/prim/mat/vectorize/apply_scalar_unary.hpp>
```

Comment:
Okay. I compared this discussion to the one on the issue tracker: https://github.com/stan-dev/math/issues/202. They look very similar and this link from the tracker ultimately helped me understand it: http://stackoverflow.com/questions/14626033/implicit-instantiation-of-undefined-template-class.